go/datas/database.go

```diff
@@ -94,8 +94,13 @@ type Database interface {
 	// Regardless, Datasets() is updated to match backing storage upon return.
 	FastForward(ds Dataset, newHeadRef types.Ref) (Dataset, error)
 
+	// ValidatingBatchStore returns the BatchStore used to read and write
+	// groups of values to the database efficiently. This interface is a low-
+	// level detail of the database that should infrequently be needed by
+	// clients.
+	ValidatingBatchStore() types.BatchStore
```

Comment:
Yup, guessed this wasn't s good approach but figured a patch was best way to start convo.

The http-client db doesn't have a chunkstore right? How can I do it in that case?

I thought about just exposing Root() and UpdateRoot()...